src/libs/migrations/CheckForPreviousReportActionID.js

```diff
@@ -0,0 +1,54 @@
+import _ from 'underscore';
+import Onyx from 'react-native-onyx';
+import Log from '../Log';
+import ONYXKEYS from '../../ONYXKEYS';
+
+/**
+ * @returns {Promise<Object>}
+ */
+function getReportActionsFromOnyx() {
+    return new Promise((resolve) => {
+        const connectionID = Onyx.connect({
+            key: ONYXKEYS.COLLECTION.REPORT_ACTIONS,
+            waitForCollectionCallback: true,
+            callback: (allReportActions) => {
+                Onyx.disconnect(connectionID);
+                return resolve(allReportActions);
+            },
+        });
+    });
+}
+
+/**
+ * Migrate Onyx data for reportActions. If the first reportAction of a reportActionsForReport
+ * does not contain a 'previousReportActionID', all reportActions for all reports are removed from Onyx.
+ *
+ * @returns {Promise<void>}
+ */
+export default function () {
+    return getReportActionsFromOnyx().then((allReportActions) => {
+        if (_.isEmpty(allReportActions)) {
+            Log.info(`[Migrate Onyx] Skipped migration CheckForPreviousReportActionID because there were no reportActions`);
+            return;
+        }
+
+        const firstReportActionID = _.keys(allReportActions)[0];
+        const firstReportAction = allReportActions[firstReportActionID];
```

Comment:
since reportActions was a complex object, I spelled out how I got the first value of first reportActionID in accordance to clean code. 
`const firstValueOfFirstReportAction = _.values(_.values(allReportActions)[0])[0];`
This did look a bit cryptic to me at first. Should I change it to 1 line?